lib/cute_pets.rb

```diff
@@ -6,16 +6,43 @@
 module CutePets
   extend self
 
-  def post_pet()
-    pet = nil
+  def fetch_pet()
     if ENV.fetch('pet_datasource').downcase == 'petfinder'
       pet = PetFetcher.get_petfinder_pet
     else
       pet = PetFetcher.get_petharbor_pet
     end
+  end
+
+  def post_pet()
+    pet = nil
+    pet = fetch_pet()
+    # Check that we haven't tweeted this pet before.
+    history_file = ENV.fetch('history_file')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.